 ERROR root [2020-10-28 00:00:36,611] - Components::removeComponentByChannel: logger : 210002, Abnormal exit! TCPPacketReceiver::processRecv(): errno=10054

  INFO root [2020-10-28 00:00:36,620] - The following logs sent to logger failed:
  WARN root [2020-10-28 00:00:36,622] - ==> [2020-10-28 00:00:36 611] TCPPacketReceiver::processPendingEvents(192.168.2.246:49801): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2020-10-28 00:00:36,623] - ==> [2020-10-28 00:00:36 618] Components::removeComponentByChannel: logger : 210002, Abnormal exit! TCPPacketReceiver::processRecv(): errno=10054

  WARN root [2020-10-28 00:00:36,694] - TCPPacketReceiver::processPendingEvents(192.168.2.246:49802): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2020-10-28 00:00:36,695] - Components::removeComponentByChannel: dbmgr : 42002, Abnormal exit! TCPPacketReceiver::processRecv(): errno=10054

  INFO root [2020-10-28 00:00:36,698] - Shutdowner::onShutdownBegin: shutting down(period=0, tickPeriod=1)
S_INFO root [2020-10-28 00:00:36,699] - onLoginAppShutDown()
  INFO root [2020-10-28 00:00:36,700] - Shutdowner::onShutdown: shutting down
  INFO root [2020-10-28 00:00:36,700] - Shutdowner::onShutdownEnd: shutting down
 DEBUG root [2020-10-28 00:00:36,705] - Script::uninstall_py_dlls(): PyInit_pyexpat
 DEBUG root [2020-10-28 00:00:36,705] - Script::uninstall_py_dlls(): PyInit__socket
 DEBUG root [2020-10-28 00:00:36,706] - Script::uninstall_py_dlls(): PyInit__ssl
 DEBUG root [2020-10-28 00:00:36,706] - Script::uninstall_py_dlls(): PyInit__hashlib
 DEBUG root [2020-10-28 00:00:36,707] - Script::uninstall_py_dlls(): PyInit_select
 DEBUG root [2020-10-28 00:00:36,707] - Script::uninstall_py_dlls(): PyInit__ctypes
 DEBUG root [2020-10-28 00:00:36,708] - Script::uninstall_py_dlls(): PyInit__elementtree
 DEBUG root [2020-10-28 00:00:36,708] - Script::uninstall_py_dlls(): PyInit_unicodedata
  INFO root [2020-10-28 00:00:36,732] - Script::uninstall(): is successfully!
 DEBUG root [2020-10-28 00:00:36,733] - ThreadPool::destroy(): starting size 2.
  INFO root [2020-10-28 20:11:47,431] - 
  INFO root [2020-10-28 20:11:47,433] - -----------------------------------------------------------------------------------------


  INFO root [2020-10-28 20:11:47,434] - Resmgr::initialize: KBE_ROOT=D:/kbengine_24pic/
  INFO root [2020-10-28 20:11:47,434] - Resmgr::initialize: KBE_RES_PATH=D:/kbengine_24pic/kbe/res/;D:/kbengine_24pic/kbengine_stone_assets/;D:/kbengine_24pic/kbengine_stone_assets/scripts/;D:/kbengine_24pic/kbengine_stone_assets/res/
  INFO root [2020-10-28 20:11:47,434] - Resmgr::initialize: KBE_BIN_PATH=D:/kbengine_24pic/kbe/bin/server/
  INFO root [2020-10-28 20:11:47,439] - NetworkInterface::initialize(EXTERNAL): bound to all interfaces with default route interface on 192.168.2.246 ( 192.168.2.246:20013 )
  INFO root [2020-10-28 20:11:47,439] - NetworkInterface::initialize(EXTERNAL): address 192.168.2.246:20013, SOMAXCONN=128.
  INFO root [2020-10-28 20:11:47,442] - NetworkInterface::initialize(INTERNAL): bound to all interfaces with default route interface on 192.168.2.246 ( 192.168.2.246:50337 )
  INFO root [2020-10-28 20:11:47,442] - NetworkInterface::initialize(INTERNAL): address 192.168.2.246:50337, SOMAXCONN=128.
  INFO root [2020-10-28 20:11:47,442] - server-configs:
  INFO root [2020-10-28 20:11:47,442] - 	internalAddr : 192.168.2.246:50337
  INFO root [2020-10-28 20:11:47,442] - 	externalAddr : 192.168.2.246:20013
  INFO root [2020-10-28 20:11:47,443] - 	componentID : 920002
  INFO root [2020-10-28 20:11:47,554] - Components::findLogger: found logger, addr:192.168.2.246:50313
  INFO root [2020-10-28 20:11:47,554] - Components::addComponent: reset orderLog, uid=15930!
  INFO root [2020-10-28 20:11:47,555] - Components::addComponent[logger], uid=15930, componentID=210002, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2020-10-28 20:11:47,558] - ---- loginapp Version: 1.2.7. ScriptVersion: 0.1.0. Protocol: A87D1428E6DE7B1742DD55BD6E5BAE8C. Config: Debug X86. Built: 19:01:57 Aug  4 2020. AppID: 920002. UID: 15930. PID: 4028 ----
  INFO root [2020-10-28 20:11:47,558] - Server : Intel(R) Core(TM) i5-10210U CPU @ 1.60GHz with 2097151kB RAM
  INFO root [2020-10-28 20:11:47,559] - ThreadPool::createThreadPool: creating  threadpool...
  INFO root [2020-10-28 20:11:47,559] - ThreadPool::createThreadPool: successfully(2), newThreadCount=1, normalMaxThreadCount=2, maxThreadCount=8
 DEBUG root [2020-10-28 20:11:47,690] - Script::install_py_dlls(): PyInit_pyexpat
 DEBUG root [2020-10-28 20:11:47,690] - Script::install_py_dlls(): PyInit__socket
 DEBUG root [2020-10-28 20:11:47,691] - Script::install_py_dlls(): PyInit__ssl
 DEBUG root [2020-10-28 20:11:47,693] - Script::install_py_dlls(): PyInit__hashlib
 DEBUG root [2020-10-28 20:11:47,694] - Script::install_py_dlls(): PyInit_select
 DEBUG root [2020-10-28 20:11:47,694] - Script::install_py_dlls(): PyInit__ctypes
 DEBUG root [2020-10-28 20:11:47,694] - Script::install_py_dlls(): PyInit__elementtree
 DEBUG root [2020-10-28 20:11:47,722] - Script::install_py_dlls(): PyInit_unicodedata
  INFO root [2020-10-28 20:11:47,750] - Script::install(): is successfully, Python=(3.4.2 (default, Aug  4 2020, 19:01:54) [MSC v.1925 32 bit (Intel)])!
  INFO root [2020-10-28 20:11:47,750] - Script::install(): KBExtra is successfully!
S_INFO root [2020-10-28 20:11:47,766] - onLoginAppReady: bootstrapGroupIndex=0, bootstrapGlobalIndex=0
  INFO root [2020-10-28 20:11:47,766] - TelnetServer server is running on port 31000
  INFO root [2020-10-28 20:11:47,939] - ---- loginapp is running ----
 DEBUG root [2020-10-28 20:11:47,939] - Components::process(): Request for the process of identity...
  WARN root [2020-10-28 20:11:54,090] - Forwarding logs to logger[192.168.2.246:50313/0/0/0]...
 ERROR root [2020-10-28 21:38:07,501] - Components::removeComponentByChannel: logger : 210002, Abnormal exit! TCPPacketReceiver::processRecv(): errno=10054

  INFO root [2020-10-28 21:38:07,532] - The following logs sent to logger failed:
  WARN root [2020-10-28 21:38:07,532] - ==> [2020-10-28 21:38:07 502] TCPPacketReceiver::processPendingEvents(192.168.2.246:50313): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2020-10-28 21:38:07,532] - ==> [2020-10-28 21:38:07 524] Components::removeComponentByChannel: logger : 210002, Abnormal exit! TCPPacketReceiver::processRecv(): errno=10054

  WARN root [2020-10-28 21:38:07,579] - TCPPacketReceiver::processPendingEvents(192.168.2.246:50314): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2020-10-28 21:38:07,579] - Components::removeComponentByChannel: dbmgr : 42002, Abnormal exit! TCPPacketReceiver::processRecv(): errno=10054

  INFO root [2020-10-28 21:38:07,595] - Shutdowner::onShutdownBegin: shutting down(period=0, tickPeriod=1)
S_INFO root [2020-10-28 21:38:07,595] - onLoginAppShutDown()
  INFO root [2020-10-28 21:38:07,595] - Shutdowner::onShutdown: shutting down
  INFO root [2020-10-28 21:38:07,595] - Shutdowner::onShutdownEnd: shutting down
 DEBUG root [2020-10-28 21:38:07,595] - Script::uninstall_py_dlls(): PyInit_pyexpat
 DEBUG root [2020-10-28 21:38:07,595] - Script::uninstall_py_dlls(): PyInit__socket
 DEBUG root [2020-10-28 21:38:07,595] - Script::uninstall_py_dlls(): PyInit__ssl
 DEBUG root [2020-10-28 21:38:07,595] - Script::uninstall_py_dlls(): PyInit__hashlib
 DEBUG root [2020-10-28 21:38:07,595] - Script::uninstall_py_dlls(): PyInit_select
 DEBUG root [2020-10-28 21:38:07,610] - Script::uninstall_py_dlls(): PyInit__ctypes
 DEBUG root [2020-10-28 21:38:07,610] - Script::uninstall_py_dlls(): PyInit__elementtree
 DEBUG root [2020-10-28 21:38:07,610] - Script::uninstall_py_dlls(): PyInit_unicodedata
  INFO root [2020-10-28 21:38:07,642] - Script::uninstall(): is successfully!
 DEBUG root [2020-10-28 21:38:07,642] - ThreadPool::destroy(): starting size 2.
  WARN root [2020-10-28 21:38:07,954] - ThreadPool::destroy(): waiting for thread(2)[0x87b618,0x87ba90,], try=1
